### PR TITLE
Test SKU dynamics sampler

### PR DESCRIPTION
**Pull request by v-heli**
Perfect  in SC scenario   sku_ dynamics_ sampler test

### CODE REVIEW

File: maro/Untitled.ipynb [hunk 1]
```diff
@@ -0,0 +1,290 @@
+{
```
Comment (Jinyu-W) [inline maro/Untitled.ipynb:1]: what's the file used for?

Never use `git add *` please

File: tests/supply_chain/test_supply_chain.py [hunk 2]
```diff
@@ -1364,6 +1368,444 @@ def test_consumer_receive_products_after_vlt_days(self) -> None:
         states = consumer_nodes[expected_frame:consumer_node_index:features].flatten().astype(np.int)
         self.assertEqual(required_quantity_2, states[IDX_RECEIVED])
 
+    def test_init_sku_dynamics_OneTimeSkuPriceDemandSampler(self):
+        """Test the reading of "store_001" SKU information of OneTimeSkuPriceDemandSampler."""
+        env = build_env("case_04", 100)
```
Comment (Jinyu-W) [inline tests/supply_chain/test_supply_chain.py:1373]: add the test data files on which the test is based

File: tests/supply_chain/test_supply_chain.py [hunk 3]
```diff
@@ -1364,6 +1368,444 @@ def test_consumer_receive_products_after_vlt_days(self) -> None:
...
+        be = env.business_engine
+        assert isinstance(be, SupplyChainBusinessEngine)
+        Store_001: FacilityBase = be.world._get_facility_by_name("Store_001")
+        configs = Store_001.configs
+        world = be.world
+        sku = OneTimeSkuPriceDemandSampler(configs, world)
+        assert isinstance(sku, OneTimeSkuDynamicsSampler)
+        # Store_ 001 has two SKUs.
+        self.assertEqual(2, len(sku._cache[0]))
+        # The price of the first item of SKU with product ID 20 is 43.11954545454545 and the demand is 25.
```
Comment (Jinyu-W) [inline tests/supply_chain/test_supply_chain.py:1383]: create a fake data file for this test (do not use the sample.csv directly), for an easier and simpler comparison.

**Review by Jinyu-W — COMMENTED** [review]
remove useless files from this PR:
- xxx.ipynb
- xxx.rar

File: maro/simulator/scenarios/supply_chain/sku_dynamics_sampler.py [hunk 4]
```diff
@@ -153,8 +153,9 @@ def _load_data_until_tick(self, tick: int) -> None:
             sku_id = self._world.sku_name2id_mapping[sku_name]
```
Comment (Jinyu-W) [inline maro/simulator/scenarios/supply_chain/sku_dynamics_sampler.py:153]: there is still a potential bug in line 136: `next(self._reader)`

File: maro/simulator/scenarios/supply_chain/sku_dynamics_sampler.py [hunk 5]
```diff
@@ -133,7 +133,11 @@ def _load_data_until_tick(self, tick: int) -> None:
         May load one more extra entry to ensure that all data at tick `tick` are loaded.
         """
         while not self._is_fp_closed and (self._latest_tick is None or self._latest_tick <= tick):
-            row = next(self._reader)  # This entry may be after `tick`. We have to process it as we already loaded it.
+            try:
+                row = next(self._reader) # This entry may be after `tick`. We have to process it as we already loaded it.
+            except StopIteration:
+                break
```
Comment (Jinyu-W) [inline maro/simulator/scenarios/supply_chain/sku_dynamics_sampler.py:139]:
```
except ...:
    self._fp.close()
    self._is_fp_closed = True
```
?

File: tests/supply_chain/test_supply_chain.py [hunk 6]
```diff
@@ -1364,6 +1368,451 @@ def test_consumer_receive_products_after_vlt_days(self) -> None:
         states = consumer_nodes[expected_frame:consumer_node_index:features].flatten().astype(np.int)
         self.assertEqual(required_quantity_2, states[IDX_RECEIVED])
 
+    def test_init_sku_dynamics_OneTimeSkuPriceDemandSampler(self):
```
Comment (lihuoran) [inline tests/supply_chain/test_supply_chain.py:1371]: In Python, it is not recommended to use upper case letters in the function name. Just use `test_init_sku_dynamics_one_time_sku_price_demand_sampler`.

Comment (lihuoran) [reply tests/supply_chain/test_supply_chain.py:1371]: Or is there any better naming methods?

File: tests/supply_chain/test_supply_chain.py [hunk 7]
```diff
@@ -1364,6 +1368,451 @@ def test_consumer_receive_products_after_vlt_days(self) -> None:
...
+        env = build_env("case_04", 600)
+        be = env.business_engine
+        assert isinstance(be, SupplyChainBusinessEngine)
+        Store_001: FacilityBase = be.world._get_facility_by_name("Store_001")
+        configs = Store_001.configs
+        world = be.world
+
+        sku_stream = StreamSkuPriceDemandSampler(configs, world)
+        assert isinstance(sku_stream, StreamSkuPriceDemandSampler)
+        # Streamskupricedemandsamples inherits streamskudynamicssampler，one tick one day
```
Comment (lihuoran) [inline tests/supply_chain/test_supply_chain.py:1421]: Use correct letter cases in comments.

**Pull request closed without merging**